src/egon/data/datasets/storages/pumped_hydro.py

```diff
@@ -0,0 +1,341 @@
+        selected = mastr.copy()
+
+        # Set capacity constraint using buffer
+        if consider_capacity:
+            selected = selected[
+                (
+                    selected.el_capacity
+                    <= row["capacity"] * (1 + buffer_capacity)
+                )
+                & (
+                    selected.el_capacity
+                    >= row["capacity"] * (1 - buffer_capacity)
+                )
+            ]
+
+        # Set geographic constraint, either choose power plants
+        # with the same postcode, city or federal state
+        if consider_location == "plz":
+            selected = selected[
+                selected.plz.astype(int).astype(str) == row["postcode"]
+            ]
+        elif consider_location == "city":
+            selected = selected[selected.city == row.city.replace("\n", " ")]
+        elif consider_location == "federal_state":
+            selected.loc[:, "federal_state"] = list_federal_states[
+                selected.federal_state
+            ].values
+            selected = selected[selected.federal_state == row.federal_state]
+
+        # Set capacity constraint if selected
+        if consider_carrier:
+            selected = selected[selected.carrier == carrier]
+
+        # If a plant could be matched, add this to matched
+        if len(selected) > 0:
+            matched = matched.append(
+                gpd.GeoDataFrame(
+                    data={
+                        "source": "MaStR scaled with NEP 2021 list",
+                        "MaStRNummer": selected.EinheitMastrNummer.head(1),
+                        "carrier": carrier,
+                        "el_capacity": row.c2035_capacity,
+                        "scenario": "eGon2035",
+                        "geometry": selected.geometry.head(1),
+                        "voltage_level": selected.voltage_level.head(1),
+                    }
+                )
+            )
+
+            # Drop matched storage units from nep
+            nep = nep.drop(index)
+
+            # Drop matched storage units from MaStR list if the location is accurate
+            if consider_capacity & consider_carrier:
+                mastr = mastr.drop(selected.index)
+
+    return matched, mastr, nep
+
+
+def get_location(unmatched):
+    """ Gets a geolocation for units which couldn't be matched using MaStR data.
+    Uses geolocator and the city name from NEP data to create longitude and
+    latitude for a list of unmatched units.
+
+    Parameters
+    ----------
+    unmatched : pandas.DataFrame
+        storage units from NEP which are not matched to MaStR but containing
+        a city information
+
+    Returns
+    -------
+    unmatched: pandas.DataFrame
+        Units for which no geolocation could be identified
+
+    located : pandas.DataFrame
+        Units with a geolocation based on their city information
+
+    """
+
+    # import geolocator
+    from geopy.geocoders import Nominatim
```

Comment:
Why is this package only imported in this function and not in the beginning of the script?

Reply:
Good question and no answer. ;-) I moved it